controllers/rabbitmqsecretengineconfig_controller.go

```diff
@@ -0,0 +1,130 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RabbitMQSecretEngineConfigReconciler reconciles a RabbitMQSecretEngineConfig object
+type RabbitMQSecretEngineConfigReconciler struct {
+	util.ReconcilerBase
+	Log            logr.Logger
+	ControllerName string
+}
+
+//+kubebuilder:rbac:groups=redhatcop.redhat.io,resources=rabbitmqsecretengineconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=redhatcop.redhat.io,resources=rabbitmqsecretengineconfigs/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the RabbitMQSecretEngineConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
+func (r *RabbitMQSecretEngineConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// Fetch the instance
+	instance := &redhatcopv1alpha1.RabbitMQSecretEngineConfig{}
+	err := r.GetClient().Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	ctx = context.WithValue(ctx, "kubeClient", r.GetClient())
+	vaultClient, err := instance.Spec.Authentication.GetVaultClient(ctx, instance.Namespace)
+	if err != nil {
+		r.Log.Error(err, "unable to create vault client", "instance", instance)
+		return r.ManageError(ctx, instance, err)
+	}
+	ctx = context.WithValue(ctx, "vaultClient", vaultClient)
+	if util.IsBeingDeleted(instance) {
+		// No resources supported for deletion.
+		return reconcile.Result{}, nil
+	}
+
+	err = r.manageReconcileLogic(ctx, instance)
+	if err != nil {
+		r.Log.Error(err, "unable to complete reconcile logic", "instance", instance)
+		return r.ManageError(ctx, instance, err)
+	}
+	
+	return r.ManageSuccess(ctx, instance)
+}
+
+func (r *RabbitMQSecretEngineConfigReconciler) manageReconcileLogic(context context.Context, instance client.Object) error {
+	log := log.FromContext(context)
+	vaultEndpoint := vaultutils.NewVaultEndpoint(instance)
+	// prepare internal values
+	err := instance.(vaultutils.VaultObject).PrepareInternalValues(context, instance)
+	if err != nil {
+		log.Error(err, "unable to prepare internal values", "instance", instance)
+		return err
+	}
+	err = vaultEndpoint.Create(context)
```

Comment:
should this be CreateorUpdate?

Reply:
Update part requires to be able to read configuration, RabbitMQ engine connection configuration is not readable, as it contain secrets as a plain text